libhoney.go

```diff
@@ -120,6 +120,30 @@ type Config struct {
 	Transport http.RoundTripper
 }
 
+// VerifyWriteKey calls out to the Honeycomb API to validate the write key, so
+// we can exit immediately if desired instead of happily sending events that
+// are all rejected.
+func VerifyWriteKey() error {
+	url := fmt.Sprintf("%s/1/team_slug", defaultBuilder.APIHost)
```

Comment:
we should use `url.Parse` and `path.Join` instead of sprintf to build the URL (same as is done in transmission)